### PR TITLE
Add email functionality

### PR DESCRIPTION
**Pull request by bkyryliuk**
Implemented email functionality.
This PR updates DAR approve flow to send notifications to the users that they gained access.
Email backend can be configured in the config.py

Email examples from testing:
screen shot 2017-01-12 at 11 55 45 am
screen shot 2017-01-12 at 11 55 54 am

Reviewers:
* @ascott 
* @mistercrunch 
* @vera-liu

### CODE REVIEW

File: superset/email.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    """
+    Send email using backend specified in EMAIL_BACKEND.
+    """
+    path, attr = configuration.get('email', 'EMAIL_BACKEND').rsplit('.', 1)
+    module = importlib.import_module(path)
+    backend = getattr(module, attr)
+    return backend(to, subject, html_content, files=files, dryrun=dryrun, cc=cc, bcc=bcc, mime_subtype=mime_subtype)
+
+
+def send_email_smtp(to, subject, html_content, files=None, dryrun=False, cc=None, bcc=None, mime_subtype='mixed'):
```
Comment (xrmx) [inline superset/email.py]: Wouldn't be easier to expect to and cc to be lists instead of doing the splitting manually?

File: superset/config.py [hunk 2]
```diff
@@ -263,6 +266,20 @@ class CeleryConfig(object):
...
+smtp_host = 'localhost'
+smtp_starttls = True
+smtp_ssl = False
+smtp_user = 'superset'
+smtp_port = 25
+smtp_password = 'superset'
+smtp_mail_from = 'superset@superset.com'
+
+try:
+    from superset_config import *  # noqa
```
Comment (mistercrunch) [inline superset/config.py]: no need to duplicate this, just move the smtp settings before the same statement above

File: superset/email.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
```
Comment (mistercrunch) [inline superset/email.py]: Can we just squeeze these functions in utils.py? Looks like modules have been multiplying in the superset namespace. Let's keep the main namespace clean.

I don't think utils.py is big yet, but if it gets there we can create a utils folder and break down the utilities in different modules

**Review by mistercrunch — DISMISSED** [review]

**Comment by mistercrunch** [comment]
LGTM

**Pull request merged**